packages/gatsby-source-wordpress/src/gatsby-browser.ts

```diff
@@ -0,0 +1,72 @@
+import type { GatsbyImageProps } from "gatsby-plugin-image"
+import * as React from "react"
+import * as ReactDOM from "react-dom"
+
+let hydrateRef
+let isFirstHydration = true
+export function onRouteUpdate(): void {
+  if (isFirstHydration) {
+    isFirstHydration = false
+    return
```

Comment:
Does this work when doing a client-side navigation to another route? Wont it only hydrate the very first page visited?

Reply:
It doens't hydrate on first pass. Only on client routes - gatsby-plugin-image is smart and takes care of the first pass without react

Reply:
Ah, got it 👍 I read the code as if it worked opposite. Thanks for the explanation!